scheduler/system_sched.go

```diff
@@ -275,7 +275,13 @@ func (s *SystemScheduler) computePlacements(place []allocTuple) error {
 	for _, missing := range place {
 		node, ok := nodeByID[missing.Alloc.NodeID]
 		if !ok {
-			return fmt.Errorf("could not find node %q", missing.Alloc.NodeID)
+			s.logger.Debug("could not find node %q", missing.Alloc.NodeID)
+			if s.failedTGAllocs == nil {
+				s.failedTGAllocs = make(map[string]*structs.AllocMetric)
+			}
+
+			s.failedTGAllocs[missing.TaskGroup.Name] = s.ctx.Metrics()
+			continue
```

Comment:
Alternatively we could just log the error here and then call `s.stack.SetNodes(nodes)` with zero nodesto allow `stack.Select` to occur and possibly populate more Metrics

Reply:
This way we get a record of which node was missing when we tried to place the alloc, though, right? That seems better than the alternative.

Reply:
Previously we'd return early but now we're collecting the metrics info (which is great!) but then continuing on. From the operator's perspective does this mean that we're going to have a partial placement?

Reply:
:+1: Makes sense.